hexa/pipelines/migrations/0046_pipelineversion_unique_pipeline_version_name.py

```diff
@@ -0,0 +1,62 @@
+import logging
+import re
+
+from django.db import migrations, models
+
+logger = logging.getLogger(__name__)
+
+
+def add_timestamp_to_duplicates(apps, schema_editor):
+    PipelineVersion = apps.get_model("pipelines", "PipelineVersion")
+    duplicates = (
+        PipelineVersion.objects.values("pipeline", "name")
+        .annotate(count=models.Count("id"))
+        .filter(count__gt=1)
+    )
+
+    for duplicate in duplicates:
+        versions = PipelineVersion.objects.filter(
+            pipeline=duplicate["pipeline"], name__contains=duplicate["name"]
+        ).order_by("created_at")
+
+        for index, version in enumerate(
+            list(versions)[:-1]
+        ):  # Skip the most recent version, keep it unchanged
+            old_name = version.name
+            version.name = f"{old_name} (v{index+1})"
+            version.save()
+            logger.info(
+                f"Updated version name from {old_name} to {version.name} for {version.pipeline}"
+            )
+
+
+def revert_timestamp_from_duplicates(apps, schema_editor):
+    PipelineVersion = apps.get_model("pipelines", "PipelineVersion")
+    versions = PipelineVersion.objects.all()
+
+    version_pattern = re.compile(r" \(v\d+\)$")
```

Comment:
Might lead to false positive cases but is it really a problem ?

Reply:
IMO I would not try to implement a reverse migration. We tend to use those when it's mandatory to transform values to make the application working after a downgrade (value types, specific set of values, ...).

Reply:
OK I will remove, thanks